app/templates/_gruntfile.js

```diff
@@ -89,7 +89,29 @@ module.exports = function(grunt) {
                 type = type === 'deferred' ? 'app/_' + type : type;
 
                 // get file for replacement
-                return grunt.file.read('components/' + type + '/' + file + '/' + file + '.html');
+                output =  grunt.file.read('components/' + type + '/' + file + '/' + file + '.html');
+
+                // check if output include some 'app placeholder'
+                match = output.match( /{(app):{([\w|\-]{0,})}}/g );
+                if( match !== null ) {
+
+                  // replace each placeholder with its accociated file content
+                  match.forEach(function(elem){
```

Comment:
is a recursive call of replacement function possible? seems like you are doing the same thing all over again.